### PR TITLE
Internal AnnotationSpec clean up.

### PR DESCRIPTION
**Pull request by sormuras**
Better place, no public API change. This PR was part of #365

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/AnnotationSpec.java [hunk 1]
```diff
@@ -226,6 +216,27 @@ public Builder addMember(String name, CodeBlock codeBlock) {
       return this;
     }
 
+    /**
+     * Delegates to {@link #addMember(String, String, Object...)}, with parameter {@code format}
+     * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
+     * the class of the given {@code value} object is not supported.
+     */
+    private Builder addValue(String memberName, Object value) {
```
Comment (swankjesse) [inline src/main/java/com/squareup/javapoet/AnnotationSpec.java]: addMember? Value has a special meaning in annotations.

Comment (sormuras) [inline src/main/java/com/squareup/javapoet/AnnotationSpec.java]: *ganrflrrrg* ...then we're back to the naming conflict discovered by you. See #365 :smile:

Comment (swankjesse) [inline src/main/java/com/squareup/javapoet/AnnotationSpec.java]: Ahhh. Ship it.

**Comment by sormuras** [comment]
Great. Now, I'll rebase #365 and leave it up to you, whether to merge or flush it...

**Comment by swankjesse** [comment]
Yeah, do keep me busy!

**Comment by sormuras** [comment]
> Well, I’ll do my very best

File: src/main/java/com/squareup/javapoet/AnnotationSpec.java [hunk 2]
```diff
@@ -226,6 +216,27 @@ public Builder addMember(String name, CodeBlock codeBlock) {
...
+      if (value instanceof Class<?>) {
+        return addMember(memberName, "$T.class", value);
+      }
+      if (value instanceof Enum) {
+        return addMember(memberName, "$T.$L", value.getClass(), ((Enum<?>) value).name());
```
Comment (sormuras) [inline src/main/java/com/squareup/javapoet/AnnotationSpec.java]: Could be replaced by
```java
return addMember(memberName, "$R", value);
```
...if only #350 was merged. :dart:

**Pull request merged**